### PR TITLE
WIP: document CI-next vision

### PR DESCRIPTION
**Pull request by wainersm**

### CODE REVIEW

File: docs/CI_next.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## Introduction
+
+The Continuous Integration (CI) of Confidential Containers (CoCo) since its release 0.1.0 has evolved without major design changes despite many problem being noticed along the way, some really slowing down the development/releases, but we kept applying fixes and small improvements for the sake of just supporting the growth of the community and the initial releases. With the emerging of the "merge to main" initiative and recent disruptive changes on the CI of Kata Containers on main branch (e.g. migration from Jenkins to Github Actions) our CI reached a major breakage point.
```
Comment (GabyCT) [inline docs/CI_next.md:3]: /s/problem/problems

File: docs/CI_next.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+On the kick-off meeting of the recently created [CoCo CI working group](https://docs.google.com/document/d/1gVuJXZzdyZcBg0Vje6n3-uL-SYdgxHzzEw9moU2ycoE) we community seemed to agreed that this is the time to re-design the CI to fix the problems that has hurt our development process. This document aims to capture the CI vision that was brought on that meeting as well as on the subsequents. On the next sections you will 
+
+## The current state of affairs
+
+But what's actually the current CoCo's CI?
+
+There are two CIs that are considered:
+
+ 1. Kata Containers CI where jenkins jobs are triggered when a pull request to kata-containers/kata-containers' CCv0 branch. Each job implement the entire workflow (from build to test) that will be described below.
```
Comment (GabyCT) [inline docs/CI_next.md:13]: s/implement/implements

File: docs/CI_next.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ 1. Operator CI where jenkins jobs are triggered when a pull request to confidential-containers/operator.
+
+### Kata Containers CI for CoCo
+
+Each Jenkins job run either on Azure VM or baremetal. It does:
```
Comment (GabyCT) [inline docs/CI_next.md:18]: s/run/runs

File: docs/CI_next.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+### Current Kata Containers CI on main
+
+The current Kata Containers CI on main fixed some issues:
+ * Build the whole stack exactly once, no more redundant builds
+
+### The problems
+
+This isn't an compreensive list but rather just the ones we will tackle on CI next:
+
+## A vision for the CI next
```
Comment (ldoktor) [inline docs/CI_next.md:43]: IIUC those 3 sections should cover the:

1. current status of Kata Containers CI
2. list of ongoing issues
3. tasks to solve the issues

am I getting this right?

File: docs/CI_next.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+It's out of scope the CI for other components of CoCo (e.g. guest-components) other than the operator and Kata Containers.
```
Comment (ldoktor) [inline docs/CI_next.md:45]: I don't understand what this sentence is about, can you please elaborate a bit?

File: docs/CI_next.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+On every step below there are problems, being the most impactful on development:
+
+ 1. (step 1) Takes too long
+    * This setup step usually take several minutes, particurly a problem when developers run locally the CI
+ 1. (step 2) Redundant builds and testing loosily binaries 
+    * Every job build almost the same components. Despicte being a waste, this increase the odds of failure due to external issues (e.g. networking failure to fetch a package)
+    * There is tested the binaries built as extracted from the tarball into the test system, rather than testing the runtime-payload installed via kata-deploy (or the operator)
+
+### Operator CI
```
Comment (ldoktor) [inline docs/CI_next.md:32]: Is this section empty on purpose? If so I'd suggest adding `TODO: describe operator CI architecture/whatever/blablabla` there so people know what should come here

File: docs/CI_next.md (hunk 3 above)
Comment (ldoktor) [reply docs/CI_next.md:18]: actually by `it does` I'd assume all the steps below should use the same `s` form so `Installs, builds and tests the dependencies in the system ...`

File: docs/CI_next.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+## The current state of affairs
+
+But what's actually the current CoCo's CI?
+
+There are two CIs that are considered:
+
+ 1. Kata Containers CI where jenkins jobs are triggered when a pull request to kata-containers/kata-containers' CCv0 branch. Each job implement the entire workflow (from build to test) that will be described below.
+ 1. Operator CI where jenkins jobs are triggered when a pull request to confidential-containers/operator.
+
+### Kata Containers CI for CoCo
```
Comment (ldoktor) [inline docs/CI_next.md:16]: One thing we discussed was also to decouple the Kata and CoCo op testing, wouldn't it make sense to create 2 documents:

* in the kata repo describing what kata CI covers and what issues should be tackled (IMO ideally via GH issues linked to that document)?
* in this repo describing what CoCo op CI covers and what issues should be covered?

In each document we should briefly mention what used to be covered, why it shouldn't be and link to the other repo for details.

**Pull request closed without merging**